docs/source/torch.compiler_custom_backends.md

```diff
@@ -0,0 +1,279 @@
+# Custom Backends
+
+## Overview
+
+`torch.compile` provides a straightforward method to enable users
+to define custom backends.
+
+A backend function has the contract
+`(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]) -> Callable`.
+
+Backend functions can be called by TorchDynamo, the graph tracing component of `torch.compile`,
+after tracing an FX graph and are
+expected to return a compiled function that is equivalent to the traced FX graph.
+The returned callable should have the same contract as the `forward` function of the original `torch.fx.GraphModule`
+passed into the backend:
+`(*args: torch.Tensor) -> List[torch.Tensor]`.
+
+In order for TorchDynamo to call your backend, pass your backend function as the `backend` kwarg in
+`torch.compile`. For example,
+
+```python
+import torch
+
+def my_custom_backend(gm, example_inputs):
+    return gm.forward
+
+def f(...):
+    ...
+
+f_opt = torch.compile(f, backend=my_custom_backend)
+
+@torch.compile(backend=my_custom_backend)
+def g(...):
+    ...
+```
+
+See below for more examples.
+
+## Registering Custom Backends
+
+You can register your backend using the `register_backend` decorator, for example,
+
+```python
+from torch._dynamo import register_backend
+
+@register_backend
+def my_compiler(gm, example_inputs):
+    ...
+```
+
+Besides the `register_backend` decorator, if your backend is in another python package, you could also register your
+backend through entry points of python package, which provides a way for a package to register a plugin for another one.
+
+**Hint**\
```

Comment:
a bit [broken](https://docs-preview.pytorch.org/pytorch/pytorch/155137/torch.compiler_custom_backends.html#registering-custom-backends) compared to [main](https://docs.pytorch.org/docs/main/torch.compiler_custom_backends.html#registering-custom-backends).
```suggestion
:::{hint}
```